sdk/ml/azure-ai-ml/tests/component/e2etests/test_component.py

```diff
@@ -371,6 +371,13 @@ def test_datatransfer_copy_mixtype_component(self, client: MLClient, randstr: Ca
             "type_contract/path.yml",
             "input_types_component.yml",
         ],
+        ids=[
```

Comment:
> Some tests used file paths during parametrization, resulting in file names including paths, so this PR also fixes that by adding ids in some parametrization calls.

I have a couple maintainability questions:

1. How obvious is it when this is done incorrectly? Parameterizing over paths seems to have been working with recordings inline in the repository, and I could imagine someone trying it again after this change goes through? 

   What breaks when this is done incorrectly? Is there any documentation we can point engineers to for this?

2. Does this only apply for paths? Or any parameterized value that has a path separator (`\` on Windows, `/` on Mac and Linux)

3. Could ID generation be handled under the hood somehow, transparently to the test writer? I'd imagine this could get out of sync somewhat easily, especially if the failure isn't obvious (see point 1).

Reply:
Hi @kdestin, thanks for leaving a comment about this! I was about to write up a spiel about how local testing works with file path names, but that pushing recordings to the assets repo fails with this setup.

However, it turns out this was only an issue for the _initial_ migration -- testing on my end showed that restoring the old recording names was fine for regular recording updates via `manage_recordings.py`. The recordings are also pulled down from the assets repo just fine.

I removed the `ids` implementations so that the only changes in this PR are now from the recording move alone.

Reply:
For your third question, though, you could programmatically modify test IDs by providing a function callback to `ids` instead of a hardcoded list. See https://docs.pytest.org/en/7.1.x/example/parametrize.html#different-options-for-test-ids.

You could keep an ID function in a utilities folder and reference it from any test that parametrizes with file paths, and do something like replacing `/` with `.` in each input value. The current system works for now, but I still do think it would be ideal to have simpler names (at the moment, looking at some recordings in file explorer can be pretty confusing because of how Windows interprets the names).